lib/std/mem.zig

```diff
@@ -1706,12 +1706,30 @@ test count {
 
 /// Returns the number of needles inside the haystack
 pub fn countScalar(comptime T: type, haystack: []const T, needle: T) usize {
+    const n = haystack.len;
+    if (n < 1) return 0;
     var i: usize = 0;
     var found: usize = 0;
 
-    while (findScalarPos(T, haystack, i, needle)) |idx| {
-        i = idx + 1;
-        found += 1;
+    if (use_vectors_for_comparison and
+        !std.debug.inValgrind() and // https://github.com/ziglang/zig/issues/17717
```

Comment:
You don't need to make this exception unless you're reading outside `haystack`, which the below code does not do (good job). Same deal for `@inComptime`.